src/ert/_c_wrappers/config/config_parser.py

```diff
@@ -13,10 +14,35 @@ class ConfigWarning(UserWarning):
     pass
 
 
+@dataclass
+class ErrorInfo:
+    line: int
+    end_line: int
+    column: int
+    end_column: int
+    message: Optional[str]
+    data: Optional[dict]
+    original_error: Optional[Exception]
+
+
 class ConfigValidationError(ValueError):
-    def __init__(self, errors: str, config_file: Optional[str] = None) -> None:
+    def __init__(
+        self,
+        errors: Optional[str],  # Old setup
+        config_file: Optional[str] = None,  # Old setup
```

Comment:
Having two different concepts, old and new, in the same class is going to become too complicate.

It seems to me that the fields you need are
```
self.errors = List[Tuple[str, LocationInfo]]
```